assets/app/components/site/Pages/pagesContainer.js

```diff
@@ -14,85 +16,65 @@ class Pages extends React.Component {
   }
 
   componentDidMount() {
-    siteActions.fetchFiles(this.props.site, this.getPath(this.props.params));
+    fetchFiles(this.props.site, this.props.params);
   }
 
   componentWillReceiveProps(nextProps) {
     const { params, site } = nextProps;
-    const nextFiles = this.getFilesByPath(site.files, this.getPath(params));
-    const files = this.getFilesByPath(site.files, this.getPath(params));
+    const nextFiles = getFilesByPath(site.files, getPath(params));
+    const files = getFilesByPath(site.files, getPath(params));
 
     if (files.length && nextFiles.length === files.length) return;
 
-    siteActions.fetchFiles(site, this.getPath(params));
-  }
-
-  getLinkFor(page, id, branch) {
-    const path = page.path;
-
-    return this.isDir(page) ?
-      `/sites/${id}/tree/${path}` : `/sites/${id}/edit/${branch}/${path}`;
+    fetchFiles(site, params);
   }
 
-  getPath(routeParams) {
-    const { splat, fileName } = routeParams;
-    let path = '';
+  render() {
+    const { site, params } = this.props;
 
-    if (splat) {
-      path = `${splat}/${fileName}`;
+    if (!site) {
+      return null;
     }
-    else if (fileName) {
-      path = fileName;
+
+    if (hasNavigationJsonContent(site)) {
+      return <NavigationJsonPageList site={ site } />;
     }
 
-    return path;
+    const path = getPath(params);
+    const files = getFilesByPath(site.files, path) || [];
+    return <PageList site={ site } files={ files }/>;
   }
+}
 
+const fetchFiles = (site, params) => {
+  siteActions.fetchFiles(site, getPath(params));
+};
 
-  getFilesByPath(files = [], startingPath = '') {
-    const isRoot = (startingPath === '');
-    const path = (!isRoot) ? `${startingPath}/` : '((?!/).)*$';
-    const startsWithPath = new RegExp(`^${path}`);
+const getPath = (routeParams) => {
```

Comment:
I think in the case of React we don't need to worry about someone calling a method directly on a container, that is a huge anti-pattern and I don't think there would be a reason for a user to do that.

I don't hate the idea of moving the checks into actions; if we did that, though, would we start moving all that sort of business logic into the actions? For example, in my file picker component, would I want to move file size verification logic into an action?